mod/dagmodifier_test.go

```diff
@@ -410,6 +412,40 @@ func testDagTruncate(t *testing.T, opts testu.NodeOpts) {
 	}
 }
 
+func TestDagModifier_Sync(t *testing.T) {
+	dserv := testu.GetDAGServ()
+	nd := dag.NodeWithData(unixfs.FilePBData(nil, 0))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dagmod, err := NewDagModifier(ctx, nd, dserv, testu.SizeSplitterGen(512))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = dagmod.Write([]byte("test1"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = dagmod.Sync()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = dagmod.Truncate(0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = dagmod.Write([]byte("test2"))
+	err = dagmod.Sync()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
What are we testing in this function? We are not reading back anything so if we didn't write at the correct offset we won't detect it.